src/js/actions/layers.js

```diff
@@ -1296,9 +1296,16 @@ define(function (require, exports) {
                 var allLayerIDs = collection.pluck(results, "ID")
                     .reduce(function (allLayerIDs, layerIDs) {
                         return allLayerIDs.concat(layerIDs);
-                    }, []);
+                    }, []),
+                    allLayers = allLayerIDs.map(function (currentVal) {
+                        return { id: currentVal }
```

Comment:
perhaps I'm splitting hairs, but you are creating this list of objects (with a single "id" property) even though the selectLayers command is expecting `Layer|Immutable.Iterable.<Layer>`